### PR TITLE
9 transitioning homepage to typescript

### LINKED ISSUE #9: Transitioning `/homepage` to TypeScript
### Description:
Components and styling for the homepage components

### Goals:
- [x] Convert all JavaScript files to TypeScript.
- [x] Ensure all components, functions, and variables have proper typings.
- [x] Resolve any TypeScript compilation errors.
- [x] Update import/export statements to TypeScript syntax.
- [x] Refactor any necessary code to comply with TypeScript best practices.
- [x] Ensure TypeScript strict mode is enabled and configured appropriately.

### Steps to Complete:
1. **Analysis:**
   - [x] Identify all JavaScript files within the directory.
   - [x] Determine the types of components, functions, and variables used.

2. **Conversion:**
   - [x] Rename `.jsx`|`.js` files to `.tsx`|`.ts`.
   - [x] Add TypeScript typings to all components, functions, and variables.

3. **TypeScript Configuration:**
   - [x] Ensure `tsconfig.json` is properly configured.
   - [x] Enable TypeScript strict mode (`strict: true`).
   - [x] Configure any necessary TypeScript options for the directory.

4. **Testing and Debugging:**
   - [x] Test the application to ensure proper functionality after conversion.
   - [x] Address and fix any TypeScript compilation errors.
   - [x] Debug any runtime issues that arise from the transition.

5. **Documentation:**
   - [x] Document any significant changes made during the transition process.
   - [ ] Update README or relevant documentation with TypeScript usage guidelines for the directory.

### Additional Notes:
[Include any additional notes or considerations relevant to this directory's transition.]

### Project:
[Jobly Transition to TypeScript](https://github.com/users/nathanrydel/projects/5)
[Frontend Transition](https://github.com/nathanrydel/react-jobly-v2/milestone/1)

### Labels:
- [x] TypeScript
- [x] Transition
- [x] Directory: homepage

### PR DESCRIPTION
**Pull request by evengene**
Closes issue [#9](https://github.com/nathanrydel/react-jobly-v2/issues/9)

### Changes
 - Update files in the `Homepage` folder to `*.tsx`, `*.ts`,
 - Downgrade `typescript` from `4.9.5` to `5.4.3` due to compatibility issues with `react-scripts`,
 - Refactor `Homepage.tsx`,
 - Create routes constant to potentially move hardcoded path names

### CODE REVIEW

File: frontend/tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,10 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es6",
     "lib": ["dom", "dom.iterable", "esnext"],
+    "paths": {
+      // "~/*": ["src/*"] // Potentially create this path aliases
```
Comment (evengene) [inline frontend/tsconfig.json]: @nathanrydel, please let me know if this is something you are open to implement, or I can remove this line all together,
thanks

Comment (nathanrydel) [reply frontend/tsconfig.json]: It will clean up the imports in other files/dirs, so I am OK with implementing it.

Thank you

Comment (nathanrydel) [inline frontend/tsconfig.json]: Uncomment this to setup the path mapping and I'll merge it once completed

**Review by nathanrydel — CHANGES_REQUESTED** [review]

**Comment by evengene** [comment]
hey @nathanrydel, 
ready for re-review,
I'll handle the paths in a separate PR, I can create an issue for it if you want

**Review by nathanrydel — APPROVED** [review]
Looks great, thank you for the help.

**Pull request merged**